### PR TITLE
[bitnami/kube-prometheus] Reorganize kubelet.servicemonitor logic

### LINKED ISSUE #12953: [bitnami/kube-prometheus] Duplicate kubelet-0|1 target, bug or working as intended?
### Name and Version

bitnami/8.0.16

### What steps will reproduce the bug?

I made sure to delete any old k8s services left in kube-system which could be causing the duplicate target as investigated here: https://github.com/prometheus-community/helm-charts/issues/614#issuecomment-1083572953

There are still two instances of kubelet-0, kubelet-1. I can see that one of the jobs references cadvisor while the other is only for kubelet. Initially I thought this was duplicated metrics which would be causing some performance impact, but is this working as intended?

Other helm charts have taken this into consideration, and removed any service monitor from appearing to have duplicates like so: https://github.com/prometheus-community/helm-charts/blob/53f1ae41247976fae3bba6b4e5acc9138f58aeca/charts/kube-prometheus-stack/values.yaml#L856-L861

### Are you using any custom parameters or values?

values.yaml

### What is the expected behavior?

Only have one kubelet-0 displayed, unless this is working as intended

image

### What do you see instead?

Duplicated entries

image

### Additional information

_No response_

### PR DESCRIPTION
**Pull request by CeliaGMqrz**
### Description of the change

 Related to this open [issue](https://github.com/bitnami/charts/issues/12953)

After checking the kubelet targets upstream and looking for the same results as in our chart, we can see that there is no duplication because each endpoint shows different metrics. However, it is true that in the [upstream](https://github.com/prometheus-community/helm-charts/blob/main/charts/kube-prometheus-stack/values.yaml#L968) you can enable and disable CAdvisor metrics independently. The only permanent endpoint should be `/metrics.`

This change reorganizes the logic related to `kubelet.serviceMonitor`. CAdvisor will be independent.

Also it has been add a `kubelet.serviceMonitor.resource` set as false has been added according to the [upstream](https://github.com/prometheus-community/helm-charts/blob/53f1ae41247976fae3bba6b4e5acc9138f58aeca/charts/kube-prometheus-stack/values.yaml#L856-L861).

### Benefits

The endpoint for cadvisor metrics can be disabled.

### Applicable issues

  - fixes #12953 

### Checklist

- [X] Chart version bumped in `Chart.yaml` according to [semver](http://semver.org/). This is *not necessary* when the changes only affect README.md files.
- [X] Variables are documented in the values.yaml and added to the `README.md` using [readme-generator-for-helm](https://github.com/bitnami-labs/readme-generator-for-helm)
- [X] Title of the pull request follows this pattern [bitnami/<name_of_the_chart>] Descriptive title
- [X] All commits signed off and in agreement of [Developer Certificate of Origin (DCO)](https://github.com/bitnami/charts/blob/main/CONTRIBUTING.md#sign-your-work)

### CODE REVIEW

File: bitnami/kube-prometheus/values.yaml [hunk 1]
```diff
@@ -1942,6 +1942,20 @@ kubelet:
...
+    ##
+    resourcePath: "/metrics/resource/v1alpha1"
+    ## @param kubelet.serviceMonitor.resourceRelabelings Metric relabeling
+    ## ref: https://github.com/prometheus-operator/prometheus-operator/blob/main/Documentation/api.md#relabelconfig
+    ##
+    resourceRelabelings: []
+    ## @param kubelet.serviceMonitor.resourceMetricRelabelings Metric relabeling
+    ## ref: https://prometheus.io/docs/prometheus/latest/configuration/configuration/#metric_relabel_configs
+    ##
+    resourceMetricRelabelings: []
```
Comment (carrodher) [inline bitnami/kube-prometheus/values.yaml:1958]:
```suggestion
    ## @param kubelet.serviceMonitor.resource Enable scraping /metrics/resource from kubelet's service
    ##
    resource: false
    ## @param kubelet.serviceMonitor.resourcePath From kubernetes 1.18, /metrics/resource/v1alpha1 renamed to /metrics/resource
    ##
    resourcePath: "/metrics/resource/v1alpha1"
    ## @param kubelet.serviceMonitor.resourceRelabelings Metric relabeling
    ## ref: https://github.com/prometheus-operator/prometheus-operator/blob/main/Documentation/api.md#relabelconfig
    ##
    resourceRelabelings: []
    ## @param kubelet.serviceMonitor.resourceMetricRelabelings Metric relabeling
    ## ref: https://prometheus.io/docs/prometheus/latest/configuration/configuration/#metric_relabel_configs
    ##
    resourceMetricRelabelings: []
```

File: bitnami/kube-prometheus/values.yaml [hunk 2]
```diff
@@ -1942,6 +1942,20 @@ kubelet:
     ## ref: https://github.com/prometheus-operator/prometheus-operator/blob/master/Documentation/api.md#endpoint
     ##
     interval: ""
+    ## @param kubelet.serviceMonitor.resource. Enable scraping /metrics/resource from kubelet's service
```
Comment (carrodher) [inline bitnami/kube-prometheus/values.yaml:1945]:
```suggestion
    ## @param kubelet.serviceMonitor.resource Enable scraping /metrics/resource from kubelet's service
```

File: bitnami/kube-prometheus/values.yaml [hunk 3]
```diff
@@ -1942,6 +1942,20 @@ kubelet:
...
+    ##
+    resource: false
+    ## @param kubelet.serviceMonitor.resourcePath. From kubernetes 1.18, /metrics/resource/v1alpha1 renamed to /metrics/resource
```
Comment (carrodher) [inline bitnami/kube-prometheus/values.yaml:1948]:
```suggestion
    ## @param kubelet.serviceMonitor.resourcePath From kubernetes 1.18, /metrics/resource/v1alpha1 was renamed to /metrics/resource
```

File: bitnami/kube-prometheus/templates/exporters/kubelet/servicemonitor.yaml [hunk 4]
```diff
@@ -64,14 +69,22 @@ spec:
...
       {{- if .Values.kubelet.serviceMonitor.cAdvisorMetricRelabelings }}
       metricRelabelings: {{- include "common.tplvalues.render" ( dict "value" .Values.kubelet.serviceMonitor.cAdvisorMetricRelabelings "context" $) | nindent 8 }}
       {{- end }}
       {{- if .Values.kubelet.serviceMonitor.cAdvisorRelabelings }}
       relabelings: {{- include "common.tplvalues.render" ( dict "value" .Values.kubelet.serviceMonitor.cAdvisorRelabelings "context" $) | nindent 8 }}
       {{- end }}
+      {{- if .Values.kubelet.serviceMonitor.resource }}
+    - port: {{ printf "%s-metrics" $scheme }}
+      scheme: {{$scheme}}
+      path: {{ .Values.kubelet.serviceMonitor.resourcePath }}
```
Comment (fmulero) [inline bitnami/kube-prometheus/templates/exporters/kubelet/servicemonitor.yaml:82]: Please include tls config:
```suggestion
      path: {{ .Values.kubelet.serviceMonitor.resourcePath }}
      {{- if .Values.kubelet.serviceMonitor.https }}
      tlsConfig:
        caFile: /var/run/secrets/kubernetes.io/serviceaccount/ca.crt
        serverName: kubernetes
        insecureSkipVerify: true
      bearerTokenFile: /var/run/secrets/kubernetes.io/serviceaccount/token
      {{- else }}
```

**Review by fmulero — APPROVED** [review]
LGTM

**Pull request merged**